### PR TITLE
Remove paragraphs from text editor

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio-designer/issues/487

Before lexical created paragraphs when press enter which we tried to visually represent as line break by removing margins and converted to line break to instances.

This could lead to a few issues.

1. When paragraph block is inside `<a>` nothing can be typed in the beginning and in the end of paragraph

2. Paragraphs can be styled by users which can break text editor visually

Here I overwritten default Enter command and insert line break instead of paragraph. To protect from external sources like paste from clipboard added transform to merge all paragraphs if move than one.

We will still have one paragraph in the root with display: inline style. Though it will cause less damage to editor with user styles.

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [x] conceptual
  - [ ] detailed
  - [x] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/canvas/features/text-editor/text-editor.tsx [hunk 1]
```diff
@@ -39,6 +49,61 @@ const AutofocusPlugin = () => {
...
+    // flatten pasted paragraphs
+    const removeNodeTransform = editor.registerNodeTransform(
+      RootNode,
+      (node) => {
+        // merge paragraphs into first with line breaks between
+        if (node.getChildrenSize() > 1) {
+          const children = node.getChildren();
+          let first;
+          for (let i = 0; i < children.length; i += 1) {
+            const p = children[i];
```
Comment (kof) [inline apps/designer/app/canvas/features/text-editor/text-editor.tsx:81]: please use full names: paragraph, index

Comment (kof) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:81]: it's readable here, but it won't be somewhere else, so lets set a standard

Comment (kof) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:81]: also please add an eslint plugin for this https://github.com/webstudio-is/webstudio-designer/issues/397

File: apps/designer/app/canvas/features/text-editor/text-editor.tsx [hunk 2]
```diff
@@ -54,20 +119,24 @@ export const TextEditor = ({
...
+  const [paragraphClassName] = useState(() => nanoid());
   const [italicClassName] = useState(() => nanoid());
 
-  // initially instance styles are applied to all nodes
-  // so not necessary to use layout effect
-  useEffect(() => {
+  useLayoutEffect(() => {
     const engine = createCssEngine();
+    // reset paragraph styles and make it work inside <a>
+    engine.addPlaintextRule(`
```
Comment (kof) [inline apps/designer/app/canvas/features/text-editor/text-editor.tsx:128]: didn't you just removed paragraphs?

Comment (kof) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:128]: oh that's for our paragraph instance

Comment (kof) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:128]: no its not, ok it seems like you are keeping the lexical paragraph and only replacing pagraphs when new line is hit

Comment (TrySound) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:128]: Yep, one paragraph is lexical requirement. Even in plain text editor.

**Review by kof — APPROVED** [review]

File: apps/designer/app/canvas/features/text-editor/text-editor.tsx [hunk 3]
```diff
@@ -39,6 +49,61 @@ const AutofocusPlugin = () => {
...
+                for (const child of paragraph.getChildren()) {
+                  first.append(child);
+                }
+                paragraph.remove();
+              }
+            }
+          }
+        }
+      }
+    );
```
Comment (rpominov) [inline apps/designer/app/canvas/features/text-editor/text-editor.tsx:96]: How does this work? How do we tell whether pasted content contains paragraphs? I would expect something like `isParagraph(node)` somewhere in this code.

Comment (TrySound) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:96]: With default configuration root contains only paragraphs. Lexical converts headings to paragraphs on paste for example. So he we just check root children which are all paragraphs.

Comment (rpominov) [reply apps/designer/app/canvas/features/text-editor/text-editor.tsx:96]: Got it. Might be a good idea to add this as a code comment

**Review by rpominov — APPROVED** [review]

**Comment by kof** [comment]
Ideally this complex thing would have some tests, but I don't know if that's easy enough to test, maybe we could look  how they test lexical internally and do the same thing?

**Comment by kof** [comment]
@TrySound should we merge this now since we are releasing?

**Pull request merged**